tests/functional/override_database/test_override_database.py

```diff
@@ -0,0 +1,215 @@
+          project.adapter.Relation.create(schema=project.test_schema, identifier=self.check_caps(project, "view_1")),
+          project.adapter.Relation.create(database=os.getenv("SNOWFLAKE_TEST_ALT_DATABASE"), schema=project.test_schema, identifier=self.check_caps(project, "view_2")),
+          project.adapter.Relation.create(schema=project.test_schema, identifier=self.check_caps(project, "view_3")),
+          project.adapter.Relation.create(database=os.getenv("SNOWFLAKE_TEST_ALT_DATABASE"), schema=project.test_schema, identifier=self.check_caps(project, "view_4"))
+      ])
+
+  def test_snwoflake_database_override(self, project):
+    self.run_database_override(project)
+    # self.delete_alt_database_relation(project)
+
+
+class BaseProjectModelOverrideSnowflake(BaseOverrideDatabaseSnowflake):
+  def run_database_override(self, project):
+        run_dbt(["seed"])
+        assert len(run_dbt(["run"])) == 4
+        self.assertExpectedRelations(project)
+
+  def assertExpectedRelations(self, project):
+        check_relations_equal_with_relations(project.adapter, [
+            project.adapter.Relation.create(schema=project.test_schema, identifier=self.check_caps(project, "seed")),
+            project.adapter.Relation.create(database=os.getenv("SNOWFLAKE_TEST_ALT_DATABASE"), schema=project.test_schema, identifier=self.check_caps(project, "view_1")),
+            project.adapter.Relation.create(database=os.getenv("SNOWFLAKE_TEST_ALT_DATABASE"), schema=project.test_schema, identifier=self.check_caps(project, "view_2")),
+            project.adapter.Relation.create(schema=project.test_schema, identifier=self.check_caps(project, "view_3")),
+            project.adapter.Relation.create(database=os.getenv("SNOWFLAKE_TEST_ALT_DATABASE"), schema=project.test_schema, identifier=self.check_caps(project, "view_4"))
+        ])
+
+
+class TestProjectModelOverrideSnowflake(BaseProjectModelOverrideSnowflake):
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "config-version": 2,
+            "vars": {
+                "alternate_db": os.getenv("SNOWFLAKE_TEST_ALT_DATABASE"),
+            },
+            "models": {
+                "database": os.getenv("SNOWFLAKE_TEST_ALT_DATABASE"),
+                "test": {
+                    "subfolder": {
+                        "database": "{{ target.database }}"
+                    }
+                }
+            },
+            "seed-paths": ["seeds"],
+            "vars": {
+                "alternate_db": os.getenv("SNOWFLAKE_TEST_ALT_DATABASE"),
+            },
+            "quoting": {
+                "database": True,
+            },
+            "seeds": {
+                "quote_columns": False,
+            }
+        }
+
+    def test_snowflake_database_override(self, project):
+      self.run_database_override(project)
+    #   self.delete_alt_database_relation(project)
```

Comment:
Idea: Toss this in a class teardown method as opposed to having it as a function call? (this can lead to weird async problems that a teardown method might avoid)

Reply:
created a clean_up method and have it set to `autouse` currently. tests are now failing within the `test_snowflake_database_override` but not leaving any hanging artifacts so lateral progress?